ledger-api/grpc-definitions/com/daml/ledger/api/v1/completion.proto

```diff
@@ -29,4 +31,50 @@ message Completion {
   // Must be a valid LedgerString (as described in ``value.proto``).
   // Optional
   string transaction_id = 3;
+
+  // The application ID that was used for the submission, as described in ``commands.proto``.
+  // Must be a valid LedgerString (as described in ``value.proto``).
+  // Optional for historic completions where this data is not available.
```

Comment:
@SamirTalwar-DA Isn't it optional for an application to set it in the submission?

Reply:
Ignore, I've checked and it's required.